sqlmesh/utils/concurrency.py

```diff
@@ -13,9 +13,32 @@
 
 
 class NodeExecutionFailedError(t.Generic[H], SQLMeshError):
-    def __init__(self, node: H):
+    def __init__(self, ex: Exception, node: H):
         self.node = node
-        super().__init__(f"Execution failed for node {node}")
+
+        node_name = ""
+        if isinstance(node, SnapshotId):
+            node_name = node.name
+        elif isinstance(node, tuple):
+            node_name = node[0]
+        self.node_name = node_name
+
+        # MSSQL returns a `tuple([int] error code, [bytes] error message)`.
```

Comment:
I feel like if it's MSSQL specific, then it should be handled in the corresponding engine and not here.